rpm/BUILT-Fix-builds-on-EPEL8-due-to-outdated-components.patch

```diff
@@ -0,0 +1,188 @@
+-authors = [
+-  {name = "Red Hat Databases team"},
+-]
+-maintainers = [
+-  {name = "Ales Nezbeda", email = "anezbeda@redhat.com"}
+-]
+-description = "Templating system/generator for distributions"
+-readme = "README.md"
+-license = {file = "LICENSE"}
+-classifiers = [
+-  "Development Status :: 5 - Production/Stable",
+-  "Programming Language :: Python :: 3",
+-  "License :: OSI Approved :: GNU General Public License v2 or later (GPLv2+)"
+-]
+-
+-[tool.setuptools.cmdclass]
+-build_py = "build_manpages.build_py"
+-install = "build_manpages.install"
+-build_manpages = "build_manpages.build_manpages"
+-
+-[project.scripts]
+-dg = "distgen.__main__:main"
+-
+-[tool.setuptools.packages.find]
+-where = ["."]
+-include = ['distgen*']
+-
+-[tool.build_manpages]
+-manpages = [
+-    "man/dg.1:object=parser:pyfile=distgen/cli_parser.py",
+-]
+diff --git a/setup.cfg b/setup.cfg
+index acd96a0..12ff987 100644
+--- a/setup.cfg
++++ b/setup.cfg
+@@ -1,8 +1,49 @@
++[metadata]
++name = distgen
++version = 2.0
++author = Red Hat Databases team
++description = Templating system/generator for distributions
++long_description = file: README.md
++license = GPLv2+
++classifiers =
++    Development Status :: 5 - Production/Stable
++    Programming Language :: Python :: 3
++    License :: OSI Approved :: GNU General Public License v2 or later (GPLv2+)
++
++[options]
++include_package_data = True
++packages = find:
++python_requires = >=3.0
++install_requires = 
++    importlib-metadata
++    distro
++    jinja2
++    pyyaml
++    setuptools
++
++[options.package_data]
++distgen = distconf/*.yaml, distconf/**/*.yaml, templates/*.tpl, templates/**/*.tpl, templates/**/**/*.tpl
++
++[options.packages.find]
++exclude =
++    rpm*
++    man*
++    build_manpages*
++
++[options.entry_points]
++console_scripts =
++    dg = distgen.__main__:main
++
++[build_manpages]
++manpages =
++    man/dg.1:object=parser:pyfile=distgen/cli_parser.py
++
+ [flake8]
+-ignore = 
+-	E402,
+-	C0116
+-max-complexity = 15
++ignore=
++    # E402 module level import not at top of file
++    E402,
++    C0116
++max-complexity=15
+ 
+ [egg_info]
+ tag_build = 
+diff --git a/setup.py b/setup.py
+new file mode 100644
+index 0000000..fe58d0a
+--- /dev/null
++++ b/setup.py
+@@ -0,0 +1,31 @@
++import sys
++
++from os import path, getcwd
++from setuptools import setup
++
++from setuptools.command.build_py import build_py
++from setuptools.command.install import install
++
++project = "distgen"
++datadir = "share"
++pkgdatadir = datadir + "/" + project
++tpldir = pkgdatadir + "/templates"
++distconfdir = pkgdatadir + "/distconf"
++
++try:
++    sys.path = [path.join(getcwd(), 'build_manpages')] + sys.path
++    from build_manpages.build_manpages import (
++        build_manpages, get_build_py_cmd, get_install_cmd)
++except:
++    print("=======================================")
++    print("Use 'git submodule update --init' first")
++    print("=======================================")
++    raise
```

Comment:
The setup.py may be made minimal IMO, we don't need the "hints for users" because those are never going to call use it.  Actually, the overall solution really raises eyebrows :) and I can't say it is worth it .... but if you really plan to do the migration now - it means you want to add this backward-compat sure, but can't you simply create a `build.py` file that is going to be always present (not added-as-patch)?

Reply:
Well, there are few more issues regarding the EPEL versions.

The version of setuptools in RHEL9 fails to parse `pyproject.toml` for metadata. This means that EPEL9 also needs `setup.cfg`. EPEL8 doesn't have `build` module, so it also needs `setup.cfg`. Furthermore, EPEL8 default python doesn't even support `importlib` so that has to be patched as well anyway.

It would be possible to use `setup.py` only, which would remove the need for such a chonky patch, but we would still need a patch for `importlib`.

I see that it's not an ideal solution, but it seems at least EPEL9 could be fixed by using newer python with newer setuptools. Will experiment with that.

Reply:
The version change is orthogonal; have you considered keeping the version as is in the previous Distgen versions?

Reply:
I realized that as well. Yes, I have considered it, but I would like to source the version only from metadata of the package - afaik u can't parse `version.py` file into `pyproject.toml`, otherwise there would be duplicate mention of version - in `version.py` and in `pyproject.toml`.